dhttp/server.go

```diff
@@ -35,6 +67,59 @@ func concatConnContext(fns ...connContextFn) connContextFn {
 type testHookContextKey struct{}
 
 // ServerConfig is a mostly-drop-in replacement for net/http.Server.
+//
+// This is better than http.Server because:
+//
+//  - It natively supports "h2c" (HTTP/2 over cleartext)
+//  - Its "h2c" support is better than http.Server with golang.org/x/net/http2/h2c.NewHandler,
+//    because it properly shuts down idle h2c connections when server.Shutdown is called, rather
+//    than allowing h2c connections to sit around forever.
+//  - It uses Context cancellation as a simple and composable shutdown mechanism; supporting
+//    dcontext hard/soft contexts, rather than awkward relationship between the Shutdown and Close
+//    methods.
```

Comment:
As noted elsewhere (my "whut?" comment), the composability of `context.Context` is not something I've internalized, so maybe it's worth discussing a bit more?